kratos/utilities/read_materials_utility.cpp

```
@@ -245,32 +286,177 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::CreateSubProperties(
+    ModelPart& rModelPart,
+    Parameters Data,
+    Properties::Pointer& pNewProperty
+    )
+{
+    KRATOS_TRY;
+
+    if (Data.Has("sub_properties")) {
+
+        auto& r_list_sub_properties = pNewProperty->GetSubProperties();
+
+        const std::size_t number_of_subproperties = Data["sub_properties"].size();
+
+        // We assign the subproperties now
+        for(std::size_t i_sub_prop=0; i_sub_prop < number_of_subproperties; ++i_sub_prop) {
+            // Copy of the current parameters
+            Parameters sub_prop = Data["sub_properties"][i_sub_prop];
+
+            // We get the adress if any
+            const std::string& r_use_existing_property = sub_prop.Has("use_existing_property") ? sub_prop["use_existing_property"].GetString() : "";
+
+            // We check if already defined
+            bool already_defined = false;
+            Properties::Pointer p_new_sub_prop = nullptr;
+            if (r_use_existing_property != "") {
+                if (r_use_existing_property.size() > 1) {
+                    if (rModelPart.HasProperties(r_use_existing_property, mesh_id)) {
+                        p_new_sub_prop = rModelPart.pGetProperties(r_use_existing_property, mesh_id);
+                        already_defined = true;
+                    }
+                } else {
+                    const IndexType property_id = std::stoi(r_use_existing_property);
+                    if (rModelPart.RecursivelyHasProperties(property_id)) {
+                        p_new_sub_prop = rModelPart.pGetProperties(property_id);
+                        already_defined = true;
+                    }
+                }
+            }
+
```

here if we say to use an existing one and it doesn't exist we should throw an error-

---

OK

---

It is done later

---

3 lines later